### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by SwartzCr**
the change to src/heuristicBlocking prevents tabs from setting themselves as an instance of 3rd party tracking by checking if tabOrigin is undefined
the other changes now block social widgets in all instances when the user hasn't explicitly set them to noaction (userGreen)

### CODE REVIEW

File: src/heuristicblocking.js [hunk 1]
```diff
@@ -630,7 +630,7 @@ var heuristicBlockingAccounting = function(details) {
   else {
     var tabOrigin = tabOrigins[details.tabId];
     // Ignore first-party requests
-    if (origin == tabOrigin)
+    if (!tabOrigin || origin == tabOrigin)
       return { };
```
Comment (cooperq) [inline src/heuristicblocking.js]: needs braces

File: src/utils.js [hunk 2]
```diff
@@ -181,14 +181,9 @@ var Utils = exports.Utils = {
...
    * check if social widget replacement functionality is enabled
-   * (TODO: actually make this return something based on a user-facing setting
    */
   isSocialWidgetReplacementEnabled: function() {
-    if (!JSON.parse(localStorage.socialWidgetReplacementEnabled)){
-      return false;
-    } else {
-      return true;
-    }
+    return JSON.parse(localStorage.socialWidgetReplacementEnabled)
```
Comment (cooperq) [inline src/utils.js]: needs semicolon

File: src/webrequest.js [hunk 3]
```diff
@@ -340,18 +340,22 @@ function getSocialWidgetBlockList(tabId) {
   // a mapping of individual SocialWidget objects to boolean values
   // saying if the content script should replace that tracker's buttons
   var socialWidgetsToReplace = {};
+  green_domains = {}
```
Comment (cooperq) [inline src/webrequest.js]: semicolon

Comment (ghostwords) [inline src/webrequest.js]: What do you guys think about including and documenting a lint tool (maybe [ESLint](https://github.com/eslint/eslint)) config? You could enforce linting as part of the workflow, but even just having shared lint settings is very helpful for getting everyone on the same page code style-wise.

**Pull request merged**